### PR TITLE
Refactor code for the browser session storage

### PR DESCRIPTION
**Pull request by cschanaj**

### CODE REVIEW

**Comment by pipboy96** [comment]
@cschanaj Is the goal here to generalize `AppliedRulesets` to universal session storage?

**Comment by cschanaj** [comment]
@pipboy96 Yes, it also avoid using multiple `Map` in the global scope for similar things.

File: chromium/background-scripts/background.js [hunk 1]
```diff
@@ -187,37 +187,60 @@ chrome.browserAction.onClicked.addListener(e => {
...
+ * A centralized storage for browsing data within the browser session.
  */
-function AppliedRulesets() {
-  this.active_tab_rules = new Map();
-  this.active_tab_main_frames = new Map();
+function BrowserSession() {
+  this.tabs = new Map();
+  this.requests = new Map();
 
   let that = this;
```
Comment (pipboy96) [inline chromium/background-scripts/background.js]: Is there any reason for this? Wouldn't just turning the next function into an arrow function make `this` a part of its closure?

**Comment by pipboy96** [comment]
Can you also turn this into a ES6 class?

**Comment by Hainish** [comment]
Thanks!

**Comment by Hainish** [comment]
This is a great refactor.

**Pull request merged**